workers/data_refinery_workers/processors/salmon.py

```diff
@@ -373,6 +396,58 @@ def _run_fastqc(job_context: Dict) -> Dict:
 
     return job_context
 
+def _get_salmon_completed_exp_dirs(job_context: Dict) -> List[str]:
+    """Return a list of directory names of experiments whose samples
+    have all been processed by `salmon quant` command.
+    """
+
+    sample = job_context['samples'][0]
+    experiments = ExperimentSampleAssociation.objects.filter(sample=sample)
+    salmon_completed_exp_dirs = []
+    salmon_cmd_str = 'salmon --no-version-check quant'
+    for experiment in experiments:
+        num_salmon_completed_samples = experiment.samples.filter(
+            results__command_executed__startswith=salmon_cmd_str).distinct().count()
+        if num_salmon_completed_samples == experiment.samples.count():
+            # Remove the last two parts from the path of job_context["original_files"]
+            # (which is "<experiment_accession_code>/raw/<filename>")
+            # to get the experiment directory name.
+            dir_tokens = job_context["original_files"][0].absolute_file_path.split('/')[:-2]
+            experiment_dir = '/'.join(dir_tokens)
+            salmon_completed_exp_dirs.add(experiment_dir)
+    return salmon_completed_exp_dirs
+
+
+def _tximport(exp_dir: str, genes_to_transcripts_path: str):
+    """Run tximport R script based on input exp_dir and the path of
+    genes_to_transcripts.txt."""
+
+    result = ComputationalResult()
+    cmd_tokens = [
+        "/usr/bin/Rscript", "--vanilla",
+        "/home/user/data_refinery_workers/processors/tximport.R",
+        "--exp_dir", exp_dir,
+        "--gene2txmap", genes_to_transcripts_path
+    ]
+    # TODO: Because this job is done at experiment level, it doesn't make
+    # sense to update job_context (which is at sample level right now).
+    # The DB model may have to be modified to associate this ComputationalResult
+    # record with the experiment (instead of samples in the experiment).
```

Comment:
I disagree that we should modify the DB model. There's a many to many relationship between Samples and ComputationalResults for exactly this use case. I think every Sample from this experiment should be connected to this result.

Reply:
One reason why I don't like this idea is because when a sample belongs to multiple experiments, then a sample will be associated with multiple tximport computational results. This seems confusing.

Reply:
Hmm, that's a good point. I'm not sure which I like less. Maybe @Miserlou has thoughts on this?

Reply:
I think we need to connect it to SOMETHING, as it is here these ComputationalResults can't even be queried for except by the `command_executed` field. I'm cool with picking whichever option is easier and going with that, opening an issue about it, and figuring out the best solution later, but I think we need to be able to find the `tximport` ComputationalResult for a given sample/experiment for these to be useful at all.

Reply:
Unless we change the DB schema, the only table that  it can associate with right now is `SampleResultAssociate`. 

Suppose a user wants to find all experiments that have been processed by `tximport`. With the current DB schema, I can think of two approaches with pseudo-code like:
- Assume we have associated the tximport ComputationalResult record with each sample:
```Experiment.filter(samples[0].results.filter(command_executed.startwith('tximport ... <experiment_accession_code>).distinct()```

- Or: 
```
for each experiment:
    ComputationalResults.objects.filter(command_executed.startwith('tximport ... <experiment_accession_code>)
Then find the experiment based on <experiment_accession_code>
```
Neither of them is very satisfying to me.

Reply:
Of course, if this is not a valid user case, then we can just link this result with the samples to keep track of it w/o modifying the DB at all.

Reply:
Another question is: In the foreseeable future, is `tximport` going to be the only analysis that is at experiment level? If it is, then we can just treat as as a special case. If we will have other experiment-level jobs, then it is worth modifying the DB. Any thoughts, @jaclyn-taroni?

Reply:
We have tried, to the extent possible, to select methods that work on single samples because it streamlines the processing greatly. I don't think I could guarantee that we won't have anything else, but we are trying to avoid it to the extent possible.

Reply:
Got it. I will associate this computational result record with all related samples for now. Although it may not help the query that much, at least it doesn't hurt. I will also add an issue to keep track of it.